### ISSUE TITLE
Permission denied to access property "invoke" with Firefox

### ISSUE THREAD

**Issue by hudson-t** [angular/protractor#4272]
Bug report
---
- Node: `7.9.0`
- Protractor: `5.1.1`
- Angular: `2.4.6`
- Firefox: `53`
- Selenium Standalone: `3.4.0`
- GeckoDriver: `0.16.1`
- Operating System: `Windows 7 (64 bit)`
---

There appears to be an issue with using the latest Firefox/Protractor/Selenium/GeckoDriver. I am not 100% where the issue is.

If I run my specs with Chrome & IE, they run perfectly and I encounter no issues.

If I run with Firefox, the spec does run to the point of loading the page, but as soon as the spec attempts to interact with the page (click/sendKeys), the following error is given:
```
 - Failed: Error while running testForAngular: Error: Permission denied to access property "invoke"
        Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
        System info: host: 'COMPUTERNAME', ip: '128.87.191.165', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.8.0_121'
        Driver info: org.openqa.selenium.firefox.FirefoxDriver
        Capabilities [{moz:profile=C:\Users\073349\AppData\Local\Temp\rust_mozprofile.g9ctTA7ltsPD, rotatable=false, timeouts={implicit=0.0, pageLoad=300000.0, script=30000.0}, pageLoadStrategy=normal, platform=ANY, specificationLevel=0.0, moz:accessibilityChecks=false, acceptInsecureCerts=false, browserVersion=53.0, platformVersion=6.1, moz:processID=11912.0, browserName=firefox, javascriptEnabled=true, platformName=windows_nt}]
        Session ID: 3e61ca40-44d8-4444-9723-c38d29d24eef
```

If I run with directConnect, I get another issue but this seems to be logged in #4253.

**Comment by hudson-t** [comment]
Possibly an issue with [Zone.js](https://github.com/angular/zone.js/issues/616). I will check this now and post my findings.

**Comment by wswebcreation** [comment]
@hudson-t,

Maybe some additional info to your bug / issue. When I use your configuration on a Mac with Firefox 53 and GeckoDriver 0.16.1 I don't have the problem (we are also on angular 2+). When I run this in the Sauce Labs cloud I with your windows configuration I have the same issue as you reported here.

I'll also test with Sauce Labs and a Mac with Firefox 53 and let you know if it then works.

**Comment by hudson-t** [comment]
@wswebcreation my problem was related to the [Zone.js bug](https://github.com/angular/zone.js/issues/616). I was on v7.7. Updating to v8.6 has resolved the issue. 

However, I am now having an issue with client-side navigation error but this is not related to the original problem so I am closing the issue!

**Comment by MythriManjunath** [comment]
I get this error message while running protractor . Failed: Error while running testForAngular: Error: Permission denied to access property "invoke" .These are the versions used.

```
node -7.5.0
protractor: 5.1.2
jasmine: 2.6.0
"zone.js": "0.8.12"
"@angular/core": "2.4.10"
"typescript": "2.2.2"
browserName=firefox
browserVersion=53.0.3

Stack:
    Error: Error while running testForAngular: Error: Permission denied to access property "invoke"
    Build info: version: '3.4.0', revision: 'unknown', time: 'unknown'
    System info: host: 'ubuntu', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.4.0-79-generic', java.version: '1.8.0_121'
    Driver info: org.openqa.selenium.firefox.FirefoxDriver
    Capabilities [{moz:profile=/tmp/rust_mozprofile.0iu5oUp31CC2, rotatable=false, timeouts={implicit=0.0, pageLoad=300000.0, script=30000.0}, pageLoadStrategy=normal, platform=ANY, specificationLevel=0.0, moz:accessibilityChecks=false, acceptInsecureCerts=false, browserVersion=53.0.3, platformVersion=4.4.0-79-generic, moz:processID=3498.0, browserName=firefox, javascriptEnabled=true, platformName=linux}]
    Session ID: 0391cb95-8c0d-452b-b2dc-a1827a6244b1
        at executeAsyncScript_.then (/usr/lib/node_modules/protractor/lib/browser.ts:945:27)
        at ManagedPromise.invokeCallback_ (/usr/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:1366:14)
        at TaskQueue.execute_ (/usr/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2970:14)
        at TaskQueue.executeNext_ (/usr/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2953:27)
        at asyncRun (/usr/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:2813:27)
        at /usr/lib/node_modules/protractor/node_modules/selenium-webdriver/lib/promise.js:676:7
        at process._tickCallback (internal/process/next_tick.js:103:7)
    From: Task: Run beforeEach in control flow
        at Object.<anonymous> (/usr/lib/node_modules/protractor/node_modules/jasminewd2/index.js:94:19)
    From asynchronous test: 
    Error
        at Suite.<anonymous> (/home/ubuntu/git/idevops/Test/demoDashboardTest/test/menu.js:2:4)
        at Object.<anonymous> (/home/ubuntu/git/idevops/Test/demoDashboardTest/test/menu.js:1:1)
        at Module._compile (module.js:571:32)
        at Object.Module._extensions..js (module.js:580:10)
        at Module.load (module.js:488:32)
        at tryModuleLoad (module.js:447:12)

```

**Comment by hudson-t** [comment]
@MythriManjunath - fix for this is in Zone v8.6+ so being your app is running v8.12, I am betting that is the issue! Try updating and then check.

**Comment by MythriManjunath** [comment]
@hudson-t Yeah..I updated to Zone v8.6, but I still get the same error.
